pkg/template/event.go

```diff
@@ -27,8 +27,13 @@ import (
 )
 
 // bodyPathVarRegex determines valid body path variables
-var bodyPathVarRegex = regexp.MustCompile(`\$\(body(.[0-9A-Za-z_-]+)*\)`)
-var headerVarRegex = regexp.MustCompile(`\$\(header(.[0-9A-Za-z_-]+)?\)`)
+// The body regular expression allows for a subset of GJSON syntax, the mininum
+// required to navigate through dictionaries, query arrays and support
+// namespaced label names e.g. tekton.dev/eventlistener
+var bodyPathVarRegex = regexp.MustCompile(`\$\(body(\.[[:alnum:]/_\-\.\\]+|\.#\([[:alnum:]=<>%!"\*_-]+\)#??)*\)`)
```

Comment:
I think that the only metacharacters that we need to escape are `*+?()|` (https://github.com/google/re2/wiki/Syntax)

Reply:
It is not accurate. It is now more accurate than it was before but a regex is not guarantee that the body is valid GJSON. It would be helpful if GJSON provided a validation function.

Reply:
Yeah, validating using regex is not ideal though it does help with some use cases. So my vote is to keep it though it is not a guarantee that the GJSON is correct.  Ideally, like @afrittoli said, the library should provide a validation function!